.github/workflows/e2e_scripts/cleanup_monitoring.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from google.cloud import monitoring_v3
+from google.cloud.monitoring_dashboard import v1
+
+project_id = ''
+project_name = 'projects/'
+
+def getProjectId():
+    """Retrieves the project id from the environment variable.
+    Raises:
+        MissingProjectIdError -- When not set.
+    Returns:
+        str -- the project name
+    """
+    project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+    if not project_id:
+        raise MissingProjectIdError(
+            'Set the environment variable ' +
+            'GOOGLE_CLOUD_PROJECT to your Google Cloud Project Id.')
+    return project_id
+
+def cleanupDashboards():
+    """ Deletes all dashboards. """
+    client = v1.DashboardsServiceClient()
+    dashboards = client.list_dashboards(project_name)
+    for dashboard in dashboards:
```

Comment:
You should make sure your script blocks until it sees all resources are actually deleted as expected. We want the main tests to run if and only if the project is verified clean

The other cleanup script runs each step inside a while loop for this reason

Reply:
The script now blocks to ensure that all resources are deleted using a while loop. The only condition to break from the while loop is the resources being deleted, let me know if it needs another way to escape (aka # of tries?)